js/bib-activity-tracker.js

```diff
@@ -1,55 +1,80 @@
-function bib_trackActivity(activityData) {
-    var requestBody = bib_constructRequestBody(activityData);
-    var httpClient = bib_constructHttpClient();
-    return httpClient.send(requestBody);
-}
-
-function bib_constructRequestBody(activityData) {
-    return JSON.stringify(activityData)
-}
-
-function bib_constructHttpClient() {
-    var url = "https://4665tudai1.execute-api.eu-west-1.amazonaws.com/dev/activities";
-    var httpClient = new XMLHttpRequest();
-    httpClient.open("POST", url, false);
-    httpClient.setRequestHeader('Content-Type', 'application/json');
-    return httpClient;
-}
-
-function bib_constructActivityData(type, sourceContentItemId, clickedContentItemId, catalogueIds, relatedContentItems, instrument) {
-    var activityData = {
-        "type": type,
-        "actor": {},
-        "object": bib_constructActivityObject(clickedContentItemId),
-        "context": bib_constructActivityContext(sourceContentItemId, catalogueIds, relatedContentItems),
-        "instrument": bib_constructActivityInstrument(instrument)
-    }
-    return activityData;
-}
-
-function bib_constructActivityInstrument(type, version, config) {
-    return {
-        "type": type,
-        "version": version,
-        "config": config
-    };
-}
-
-function bib_constructActivityObject(clickedContentItemId) {
-    return [["contentItemId", clickedContentItemId]];
-}
-
-function bib_constructActivityContext(sourceContentItemId, catalogueIds, relatedContentItems) {
-    var context = [];
-    context.push(["sourceHref", window.location.href]);
-    context.push(["sourceContentItemId", sourceContentItemId]);
-    context = _.reduce(relatedContentItems, function (context, contentItem) {
-        context.push(["recommendations.contentItemId", contentItem.contentItemId])
-        return context;
-    }, context);
-    context = _.reduce(catalogueIds, function (context, catalogueId) {
-        context.push(["recommendations.catalogueId", catalogueId]);
-        return context;
-    }, context);
-    return context;
-}
+"use strict";
+
+(function() {
+  var isNodeJS = false;
+  var XMLHttpRequestNodeJS = false;
+
+  // support for NodeJS, which doesn't support XMLHttpRequest natively
+  if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') {
+    isNodeJS = true;
+    XMLHttpRequestNodeJS = require('xmlhttprequest').XMLHttpRequest;
+  }
+
+  var BibblioActivity = {
+    track: function(activityData) {
+      var requestBody = BibblioActivity.constructRequestBody(activityData);
+      var httpClient = BibblioActivity.constructHttpClient();
+      return httpClient.send(requestBody);
+    },
+
+    constructRequestBody: function(activityData) {
+      return JSON.stringify(activityData);
+    },
+
+    constructHttpClient: function() {
+      var url = "https://4665tudai1.execute-api.eu-west-1.amazonaws.com/dev/activities";
```

Comment:
Good point - was in there like that originally, but you're right. Will change it in our "-source" repo and we can just "git rm" it from this repo if we don't want it public yet :)